src/livekit/openIDSFU.ts

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2023 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { MatrixClient } from "matrix-js-sdk";
+import { logger } from "matrix-js-sdk/src/logger";
+
+import { Config } from "../config/Config";
+
+export interface SFUConfig {
+  url: string;
+  jwt: string;
+}
+
+export async function getSFUConfigWithOpenID(
+  client: MatrixClient,
+  roomName: string
+): Promise<SFUConfig> {
+  const openIdToken = await client.getOpenIdToken();
+  logger.debug("Got openID token", openIdToken);
+
+  const livekitCfg = Config.get().livekit;
+
+  if (!livekitCfg?.livekit_service_url) {
+    throw new Error("No livekit service URL defined");
+  }
```

Comment:
We probably want to move this part outside of the `getSFUConfigWithOpenID`, making the URL a mandatory parameter of a function. This way we ensure that the function can only be called with a "valid input parameters".

So far we've been using the `Config` as a global variable in different places. I think it would be easier if we read the config once (when the app is initialized) and then just pass the right values to the functions. I think for the sake of this PR it would be sufficient to just move this logic to the higher level component that calls the `getSFUConfigWithOpenID()` (similar to how it was before).

Rationale:
1. Reading/accessing the config file does not seem to belong to the responsibilities of the `getSFUConfigWithOpenID()` function. From my perspective, explicitly passing the configuration for such nice and clean 'library-looking' functions that have a single responsibility would be a better choice 🙂 
2. I imagine that if LiveKit is not configured, the branches of code that are responsible for the LiveKit initialization won't be called at all.

What do you think?

Reply:
I thought I replied to this but apparently not. Either way, yes, seems reasonable - done.